From 44b1f9ac9088a87572274f16dba70b99b1b055c8 Mon Sep 17 00:00:00 2001
From: "hongbo.wang" <hongbo.wang@nxp.com>
Date: Mon, 17 Jan 2022 16:13:00 +0800
Subject: [PATCH 2/2] igh_ethercat: fix calltrace issue caused by 'ethercatcal
 stop' when nservo_run is running

Using THIS_MODULE is ok, then if nservo_run is running, will report:
rmmod: ERROR: Module ec_dpaa1 is in use

it can avoid kernel calltrace.

Signed-off-by: hongbo.wang <hongbo.wang@nxp.com>
Upstream-Status: Pending
---
 devices/dpaa1/dpaa1-ethercat.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/devices/dpaa1/dpaa1-ethercat.c b/devices/dpaa1/dpaa1-ethercat.c
index 0a7a619..45b9592 100644
--- a/devices/dpaa1/dpaa1-ethercat.c
+++ b/devices/dpaa1/dpaa1-ethercat.c
@@ -7,12 +7,10 @@
 
 static void *ec_dpaa_netdev_init(struct net_device *net_dev)
 {
-	struct module *module = NULL;
 	ec_device_t *ecdev = NULL;
 	int ret = 0;
 
-	module = ec_dpaa_get_module();
-	ecdev = ecdev_offer(net_dev, ec_dpaa_poll, module);
+	ecdev = ecdev_offer(net_dev, ec_dpaa_poll, THIS_MODULE);
 	if (ecdev) {
 		ret = ecdev_open(ecdev);
 		if (ret) {
-- 
2.25.1

